core/crypto/src/signature.rs

```diff
@@ -668,6 +668,15 @@ pub enum Signature {
     SECP256K1(Secp256K1Signature),
 }
 
+impl Hash for Signature {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        match self {
+            Signature::ED25519(sig) => sig.to_bytes().hash(state),
```

Comment:
Unfortuate that we need to by hand this, but seems like that's just the current state: https://github.com/dalek-cryptography/ed25519-dalek/issues/52